activities/views/compose.py

```diff
@@ -28,6 +28,19 @@ class form_class(forms.Form):
             widget=forms.Textarea(
                 attrs={
                     "placeholder": "What's on your mind?",
+                    "_": f"""
+                    on load or input
+                        set characters to my.value.trim().length
+                        put { Config.system.post_length } - characters into #character-counter
```

Comment:
It looks working but I'm not confident this is the correct way to get `post_length`.

Reply:
So this is the right place to pull it from, but the wrong _time_ to pull it; you're doing this in a place that is run when the module is loaded, but it should be done each request instead (this is why the tests are failing).

You will, instead, have to either add this attribute when the form is created in its __init__, or move it into the template with a special override for this specific textarea.

I'm happy to do this for you if you'd like.

Reply:
Thanks for taking a look and giving hints🙂 Hmm, it seems like I need to understand the lifecycle of these components better. I'll try to find a right way again tomorrow.

Reply:
@andrewgodwin Ah sorry, I misunderstood your last comment when I read it at midnight. As it still seems complicated for me to amend the change, could you fix the error if you can? Thanks.

Reply:
Sure thing - I'll just merge this in as-is and then immediately fix it, rather than pushing changes to your branch (I'm feeling lazy this morning).

Thanks for figuring out the hyperscript!

Reply:
No problem at all and thanks! Yeah, takahē is how I knew the hyperscript at first but writing hyperscript is a kind of fun. It reminds me to write AppleScript :)